cli/module_registry.go

```diff
@@ -98,9 +98,18 @@ func CreateModuleAction(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	// Check to make sure the user doesn't accidentally overwrite a module manifest
-	if _, err := os.Stat(defaultManifestFilename); err == nil {
-		return errors.New("another module's meta.json already exists in the current directory. Delete it and try again")
+
+	// If a meta.json exists in the current directory and it matches the name and public-namespace,
+	// create the module and update it. If it does not match, give an error telling the user about the mismatch.
+	modManifest, err := loadManifest(defaultManifestFilename)
+	if err == nil {
+		manifestModuleID, err := parseModuleID(modManifest.ModuleID)
+		if err != nil ||
+			manifestModuleID.name != moduleNameArg ||
+			(manifestModuleID.prefix != orgIDArg && manifestModuleID.prefix != publicNamespaceArg) {
+			return errors.Errorf("another module's meta.json (%q) already exists in the current directory. Delete it and try again",
```

Comment:
not totally sure what the user hint should be here

from the workflow you gave in the PR description, they might be trying to adapt a sample meta.json, so 'delete it' is not always the right call

guessing you have a better understanding of the user flow here than I do, so I'm just flagging this, your call what to do

Reply:
Agreed -- it is a weird hint. That was the error message from before (when we didn't support this edge case args-match case). 

The reason I proposed that they delete it is that they are guaranteed to arrive at a working state if they delete and recreate instead of telling them about how to modify the manifest module_id which could be error prone and might lead to more frustration than just recreating the file.

Thoughts?

Reply:
works for me

Reply:
May I suggest: "A different module '%q' exists in the meta.json of the current directory. Either delete the meta.json or modify the name, and try again."